galaxy/api/serializers/event.py

```diff
@@ -0,0 +1,124 @@
+# (c) 2012-2018, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import copy
+import logging
+
+from galaxy import constants
+from galaxy.main import models
+
+from rest_framework import serializers as drf_serializers
+
+from django.core.urlresolvers import reverse
+
+from . import serializers
+
+
+__all__ = (
+    'EventSerializer',
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ValidSearchType(object):
+    choices = []
+
+    def __init__(self):
+        for choice in constants.EventType.choices():
+            self.choices.append(choice[0])
+
+    def __call__(self, value):
+        if value not in self.choices:
+            message = '%s is not a valid choice.' % value
+            raise drf_serializers.ValidationError(message)
+
+
+def clean_search_data(data):
+    return {
+        'search_params': data.get('search_params', {}),
+        'search_results': data.get('search_results', 0),
+        'next_page_clicks': data.get('next_page_clicks', 0),
+        'prev_page_clicks': data.get('prev_page_clicks', 0),
+        'results_clicked': data.get('results_clicked', []),
+        'repositories_clicked': data.get('repositories_clicked', []),
+        'content_items_clicked': data.get('content_items_clicked', []),
+        'last_page_size': data.get('last_page_size', 0)
+    }
+
+
+def append_list_to_list(source_data, target_data, label):
```

Comment:
It's unclear what this function does. Docstring or some explanation is required.
Function name indicates common used util function, but instead it implements some context specific business logic.